elasticdl_preprocessing/layers/hashing.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import absolute_import, division, print_function
+
+import tensorflow as tf
+
+
+class Hashing(tf.keras.layers.Layer):
+    """Distribute categorical feature values into a finite number of buckets
+    by hashing.
+
+    This layer converts a sequence of int or string to a sequence of int.
+    output_id = Hash(input_feature_string) % num_bins for string type input.
+    For int type input, the layer converts the value to string and then
+    processes it by the same formula. TensorFlow 2.2 has developed
+    `tf.keras.layers.preprocessing.Hashing` but not released it yet. So the
+    layer is a simple temporary version.
+    https://github.com/tensorflow/tensorflow/blob/r2.2/tensorflow/python/keras/layers/preprocessing/hashing.py
+
+    Note that the TensorFlow version with the layer must be greater than 2.0.0.
+
+    Example:
+    ```python
+    layer = Hashing(num_bins=3)
+    inp = np.asarray([['A'], ['B'], ['C'], ['D'], ['E']])
+    layer(inp)
+    ```
+    The output will be `[[1], [0], [1], [1], [2]]`
+
+    Arguments:
+        num_bins: Number of hash bins.
+        **kwargs: Keyword arguments to construct a layer.
+
+    Input: A string, int32 or int64 `tf.Tensor`,
+        `tf.SparseTensor` or `tf.RaggedTensor`
+
+    Output: An int64 tensor with the same type as input.
```

Comment:
An int64 tensor with the same type as input
=>
An int64 tensor with the same shape as input